tests/plugins/dut_monitor/pytest_dut_monitor.py

```diff
@@ -0,0 +1,358 @@
+                if process_consumption >= thresholds["cpu_process"]:
+                    if process_name not in process_overused:
+                        process_overused[process_name] = []
+                    # Collect list of CPU utilization for specific process if CPU utilization exceeds threshold
+                    process_overused[process_name].append((timestamp, process_consumption))
+
+        # Handle measurements per process
+        if process_overused:
+            for process_name, process_consumption in process_overused.items():
+                timestamps = []
+                process_sum = 0
+                for m_id, m_value in enumerate(process_consumption):
+                    t_stamp = datetime.strptime(m_value[0], t_format)
+                    process_sum += m_value[1]
+                    if not timestamps:
+                        timestamps.append(t_stamp)
+                        continue
+                    if (2 <= (t_stamp - timestamps[-1]).total_seconds() <= 3):
+                            timestamps.append(t_stamp)
+                            if m_id == (len(process_consumption) - 1):
+                                fail_msg += handle_process_measurements(p_name=process_name, t_first=timestamps[0], t_last=timestamps[-1], p_average=process_sum / len(timestamps))
+                    else:
+                        fail_msg += handle_process_measurements(p_name=process_name, t_first=timestamps[0], t_last=timestamps[-1], p_average=process_sum / len(timestamps))
+                        timestamps = []
+                        process_sum = 0
+
+        # Calculate average CPU utilization
+        if (total_sum / len(cpu_meas)) > thresholds["cpu_total_average"]:
+            fail_msg += "\n> Average CPU consumption during test run {}; Threshold - {}\n".format(total_sum / len(cpu_meas), thresholds["cpu_total_average"])
+
+        if fail_msg:
+            raise CPUThresholdExceeded(cpu_thresholds + fail_msg)
+
+
+class DUTMonitorClient(object):
+    """
+    DUTMonitorClient object establish SSH connection with DUT. Keeps SSH connection with DUT during full test run.
+    Available features:
+        - start/stop hardware resources monitoring on DUT
+        - automatically restart monitoring script on the DUT in case of lose network connectivity (device reboot, etc.)
+    """
+    def __init__(self, host, user, password):
+        self.running = False
+        self.user = user
+        self.password = password
+        self.host = host
+        self.init()
+        self.run_channel = None
+        self._thread = threading.Thread(name="Connection tracker", target=self._track_connection)
+        self._thread.setDaemon(True)
+        self._thread.start()
+
+    def _track_connection(self):
+        """
+        @summary: Track network connectivity. Reestablish network connection in case of drop connection
+        """
+        while True:
+            try:
+                self.ssh.exec_command("ls", timeout=5)
```

Comment:
The "ls" command would trigger IO operation. Suggest to replace it with a bash command that consumes less resource, for example `true`.

Reply:
Updated. Thank you.